drivers/sensor/ntc_thermistor/ntc_thermistor_calc.c

```diff
@@ -79,18 +38,27 @@ int ntc_compensation_compare_ohm(const struct ntc_type *type, const void *key, c
  */
 static void ntc_lookup_comp(const struct ntc_type *type, unsigned int ohm, int *i_low, int *i_high)
```

Comment:
This looks like my original implementation :-)  

The Zephyr community advised me to use the bsearch for space optimization.  The compare function pointer here should find the lower key where the value falls between.  I don't think this change is needed. See
https://github.com/zephyrproject-rtos/zephyr/pull/54947#pullrequestreview-1335999939 and https://pastebin.com/prpRVmjn

Reply:
Oh, I wasn't aware that the bsearch() approach was requested during review. Thanks for the link.

However, I don't understand how this is space optimization. The comparison function still takes up code.
I compared the generated code with and without this change (ARM build):
```
without this change:
  ntc_compensation_compare_ohm(): 62 bytes
  ntc_get_temp_mc(): 114 bytes
  compare_ohm_DT_DRV_COMPATx(): 16 bytes (per instance)
with this change:
  ntc_get_temp_mc(): 132 bytes
```
Am I missing something?

Reply:
@cfriedt - looks like with implementing our own binary search algorithm we're saving space over using `bsearch`.  Any concerns with removing this?

Reply:
There are a lot of changes happening in this PR. It would be ideal to make an apples to apples comparison.

One aspect of using bsearch is that the search itself is less intermixed with other code that tweaks this or that. So it's maybe a bit more obvious what is happening. So maybe code is a bit more maintainable / less bug prone?

Also, if 5 different drivers used bsearch instead of implementing their own bsearch, then there is commonality savings.

It is not very difficult to reduce code size for one custom use case without bsearch of course.

E.g. for small arrays, the code size could be tiny and a linear search would be faster

Reply:
> One aspect of using bsearch is that the search itself is less intermixed with other code that tweaks this or that. So it's maybe a bit more obvious what is happening and code is maybe a bit more maintainable / less bug prone?

IMHO, this comparison function is much more complicated and error prone than a 7-line binary search implementation. 

> Also, if 5 different drivers used bsearch instead of implementing their own bsearch, then there is commonality savings.

That would happen if they wanted the find-exact behavior (and in that case yes, I agree using bsearch() is a good idea). But, if they want the find-largest-not-greater-than behavior, they would all have to implement the (again, IMO) overly complicated comparison function that "tricks" bsearch() into the other behavior.

> It is not very difficult to reduce code size for one custom use case without bsearch of course.

My intent with this change isn't to optimize code size (the size comparison I made was to question the original explanation for this approach), but to make the code more understandable than the original approach. If you disagree, I can keep the old one.

> E.g. for small arrays, the code size could be tiny and a linear search would be faster

Actually, is there a reason for not doing that? Are the compensation tables ever going to be really big?

Reply:
I'm not a sensor maintainer, so will defer to them.